### PR TITLE
[fix] Use Enum value when logging hyper-parameters

### PR DESCRIPTION
**Pull request by antoinebrl**
# What does this PR do?

When logging hyper-parameters only built-in types were supported. Otherwise the class name was used. This cause a loss of information when using enum. For example when logging precision `precision: Precision`.

This contribution takes the value field if the object is an Enum. Resulting in `precision: amp_fp16`

### CODE REVIEW

**Review by eracah — APPROVED** [review]
Nice! Thanks for the contribution, @antoinebrl ! LGTM

**Comment by antoinebrl** [comment]
Sorry there was a formatting issue after I added an import. This should be solved now!

File: tests/utils/test_autolog_hparams.py [hunk 1]
```diff
@@ -6,10 +6,16 @@
...
 from composer.trainer import Trainer
-from composer.utils import (convert_flat_dict_to_nested_dict, convert_nested_dict_to_flat_dict, extract_hparams,
-                            using_torch_2)
+from composer.utils import (
+    StringEnum,
+    convert_flat_dict_to_nested_dict,
+    convert_nested_dict_to_flat_dict,
+    extract_hparams,
+    using_torch_2,
+)
```
Comment (mvpatel2000) [inline tests/utils/test_autolog_hparams.py:18]:
```suggestion
from composer.utils import (StringEnum, convert_flat_dict_to_nested_dict, convert_nested_dict_to_flat_dict,
                            extract_hparams, using_torch_2)
```

File: tests/utils/test_autolog_hparams.py [hunk 2]
```diff
@@ -42,6 +48,9 @@ class Bar:
         def __init__(self):
             self.local_hparams = {'m': 11}
 
+    class Baz(StringEnum):
+        A = "abc"
```
Comment (mvpatel2000) [inline tests/utils/test_autolog_hparams.py:52]:
```suggestion
        A = 'abc'
```

**Review by mvpatel2000 — APPROVED** [review]

**Pull request merged**